app/[locale]/(non-interactive)/feedback/page.tsx

```diff
@@ -0,0 +1,342 @@
+"use client";
+import { Button } from "@/components/ui/button";
+import NextButton from "@/components/ui/nextButton";
+import Image from "next/image";
+import { useRouter } from "next/navigation";
+import { FormEvent, useState } from "react";
+import "./styles.css";
+import SubmittedPage from "./submittedPage";
+
+type Props = {};
+
+const Page = (props: Props) => {
+  const [rating, setRating] = useState(0);
+  const [comment, setComment] = useState("");
+  const [submitted, setSubmitted] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const isEmpty = rating === 0 && comment.trim() === "";
+  const router = useRouter();
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (isSubmitting) return;
+    if (isEmpty) {
+      router.push("/end");
+      return;
+    }
+
+    setIsSubmitting(true);
+
+    fetch(`${process.env.NEXT_PUBLIC_WEB_URL}/api/feedback`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        rating,
+        comment,
+      }),
+    });
```

Comment:
This fetch is not `await`-ed so the process finishes submission without knowing whether the request going alright or not.

tl;dr: because you did not await so next line runs right after the fetch request is sent.